camel-sap/org.fusesource.camel.component.sap/src/org/fusesource/camel/component/sap/util/RfcUtil.java

```diff
@@ -1345,4 +1345,220 @@ public static String checkForKeywordLiterals(String value) {
+	 */
+	private static BigDecimal convertJcoDecimalDefaultValue(String jcoDefaultDecimalValue) {
+		if (jcoDefaultDecimalValue == null) {
+			return null;
+		}
+		try {
+			return new BigDecimal(jcoDefaultDecimalValue);
+		} catch (NumberFormatException e) {
+			return null;
+		}
+	}
+	
+	/**
+	 * Converts the JCo string representation for the default value of a date
+	 * type list parameter field into a Date.
+	 * 
+	 * @param jcoDefaultDateValue
+	 *            - the JCo string representation for the default value of a
+	 *            date list parameter
+	 * @return The Date default value.
+	 */
+	private static Date convertJCoDateDefaultValue(String jcoDefaultDateValue) {
+		if (jcoDefaultDateValue == null) {
+			return null;
+		}
+		jcoDefaultDateValue = stripSingleQuotePrefixSuffix(jcoDefaultDateValue);
+		return convertSapDateStringToDate(jcoDefaultDateValue);
+	}
+	
+	/**
+	 * Converts the JCo string representation for the default value of a time
+	 * type list parameter field into a Date.
+	 * 
+	 * @param jcoDefaultTimeValue
+	 *            - the JCo string representation for the default value of a
+	 *            time list parameter
+	 * @return The Date default value
+	 */
+	private static Date convertJCoTimeDefaultValue(String jcoDefaultTimeValue) {
+		if (jcoDefaultTimeValue == null) {
+			return null;
+		}
+		jcoDefaultTimeValue = stripSingleQuotePrefixSuffix(jcoDefaultTimeValue);
+		return convertSapTimeStringToDate(jcoDefaultTimeValue);
+	}
+	
+	/**
+	 * Converts the JCo string representation for the default value of a float
+	 * type list parameter field into a Double.
+	 * 
+	 * @param jcoDefaultFloatValue
+	 *            - the JCo string representation for the default value of a
+	 *            float list parameter
+	 * @return The Double default value.
+	 */
+	private static Double convertJCoFloatDefaultValue(String jcoDefaultFloatValue) {
+		if (jcoDefaultFloatValue == null) {
+			return null;
+		}
+		jcoDefaultFloatValue = stripSingleQuotePrefixSuffix(jcoDefaultFloatValue);
+		try {
+			return Double.valueOf(jcoDefaultFloatValue);
+		} catch (NumberFormatException e) {
+			return null;
+		}
+	}
+	
+	/**
+	 * Convert the JCo string representation for the default value of a binary type
+	 * list parameter field into a byte array.
+	 * 
+	 * @param jcoDefaultBinaryValue
+	 *            - the JCo string representation for the default value of a
+	 *            binary list parameter
+	 * @return The byte array default value.
+	 */
+	private static byte[] convertJCoBinaryDefaultValue(String jcoDefaultBinaryValue) {
+		if (jcoDefaultBinaryValue == null) {
+			return null;
+		}
+		jcoDefaultBinaryValue = stripSingleQuotePrefixSuffix(jcoDefaultBinaryValue);
+		try {
+			int length = jcoDefaultBinaryValue.length();
+			byte[] binaryData = new byte[length / 2];
+			for (int i = 0; i< length; i += 2) {
+				binaryData[i / 2] = (byte) ((Character.digit(jcoDefaultBinaryValue.charAt(i), 16) << 4) + 
```

Comment:
@apupier this is converting the JCo string representation of a byte array (a Hex String prefixed and suffixed with a single quote) into the equivalent byte array. Good idea to add a unit test for this. Thanks!